src/landscape.jl

```diff
@@ -0,0 +1,20 @@
+
+rescale!(mat) = (mat .- minimum(mat)) ./ (maximum(mat) - minimum(mat))
+
+function mask!(array, maskarray) 
+    for i in eachindex(array, maskarray)
+        array[i] = maskarray[i] == 0 ? array[i] : missing
```

Comment:
@tpoisot what do you think here? this implementation causes problems if the type of `array` does not include `missing`. But restricting to `NaN` requires us to have `Float64`.  How to make this general?

Reply:
This is the biggest problem with `SimpleSDMLayers` at the moment as well (we use `nothing` internally). I think it is safe to assume that the landscapes here are going to floating points, so using `NaN` (or `NaN32` etc) is not too bad in this specific case. The other solution is to go full sparse, but this assuming the masked part is going to be small it's almost never going to be worth it.

Reply:
In fact I'd go even further and assume that the mask should be a matrix of `AbstractBool`, and not something where we need to look at the `zero` of the type.

Reply:
~~~julia
function mask!(array, maskarray)
    array[findall(.!maskarray)] .= NaN
end
~~~

?

Reply:
no that allocates a temporary array

Reply:
but without the `findall` I guess it should work?